src/mixins/createFormLabelMixin.ts

```diff
@@ -0,0 +1,220 @@
+	 */
+	label?: string | LabelProperties;
+
+	/**
+	 * Checked/unchecked HTML attribute
+	 */
+	checked?: boolean;
+
+	/**
+	 * ID of an element that provides more descriptive text
+	 */
+	describedBy?: string;
+
+	/**
+	 * Indicates the value entered in the form field is invalid
+	 */
+	invalid?: boolean;
+
+	/**
+	 * Maximum number of characters allowed in the input
+	 */
+	maxLength?: number | string;
+
+	/**
+	 * Minimum number of characters allowed in the input
+	 */
+	minLength?: number | string;
+
+	/**
+	 * Controls whether multiple values are allowed (applies to email, file, and select inputs)
+	 */
+	multiple?: boolean;
+
+	/**
+	 * Placeholder text
+	 */
+	placeholder?: string;
+
+	/**
+	 * Allows or prevents user interaction
+	 */
+	readOnly?: boolean;
+
+	/**
+	 * Whether or not a value is required
+	 */
+	required?: boolean;
+}
+
+/**
+ * Form Label Mixin
+ */
+export interface FormLabelMixin {
+	/**
+	 * A function that generates node attribtues for the child form field
+	 */
+	getFormFieldNodeAttributes: () => VNodeProperties;
+
+	/**
+	 * Override the default getNode function on createWidgetBase to return child input wrapped in a <label>
+	 */
+	getNode(): DNode;
+}
+
+/**
+ * Form Label
+ */
+export type FormLabel = Widget<FormLabelMixinProperties> & FormLabelMixin;
+
+/**
+ * Form Label Factory interface
+ */
+export interface FormLabelMixinFactory extends ComposeFactory<FormLabelMixin, WidgetOptions<WidgetState, FormLabelMixinProperties>> {}
+
+/**
+ * Default settings for form labels
+ */
+const labelDefaults = {
+	content: '',
+	position: 'after',
+	hidden: false
+};
+
+const createFormLabelMixin: FormLabelMixinFactory = compose({
+	getFormFieldNodeAttributes(this: FormLabel): VNodeProperties {
+		const { properties, type } = this;
+		const attributeKeys = Object.keys(properties);
+
+		if (type) {
+			attributeKeys.push('type');
+		}
+
+		const allowedAttributes = ['checked', 'describedBy', 'disabled', 'invalid', 'maxLength', 'minLength', 'multiple', 'name', 'placeholder', 'readOnly', 'required', 'type', 'value'];
+		const nodeAttributes: any = {};
+
+		for (const key of allowedAttributes) {
+
+			if (attributeKeys.indexOf(key) === -1) {
+				continue;
+			}
+			else if (key === 'type') {
+				nodeAttributes.type = type;
+			}
+			else if (key === 'readOnly' && properties.readOnly) {
+				nodeAttributes.readonly = 'readonly';
+				nodeAttributes['aria-readonly'] = true;
+			}
+			else if (key === 'invalid') {
+				nodeAttributes['aria-invalid'] = properties.invalid;
+			}
+			else if (key === 'describedBy') {
+				nodeAttributes['aria-describedby'] = properties.describedBy;
+			}
+			else if ((key === 'maxLength' || key === 'minLength' || key === 'checked') && typeof properties[key] !== 'string') {
+				nodeAttributes[key.toLowerCase()] = '' + properties[key];
+			}
+			else {
+				nodeAttributes[key.toLowerCase()] = properties[key];
+			}
+		}
+
+		return nodeAttributes;
+	},
+
+	getNode(this: FormLabel): DNode {
```

Comment:
I was thinking that to do the `tag` logic it might be better if we leverage the `properties:changed` event.

It would mean a small refactor to base this mixin using `createdEvented` in order to attach to the event, similar to https://github.com/dojo/widgets/blob/master/src/mixins/themeable.ts?

In the `initialize` function there would need to be something like 

```ts
instance.own(instance.on('properties:changed', (evt: PropertiesChangeEvent<FormLabelMixin, FormLabelMixinProperties>) => {
    if (includes(evt.changedPropertyKeys, 'label')) {
        instance.tag = 'label';
    }
});

// initialise for the instantiation
if (instance.properties.label) {
    instance.tag = 'label';
}
```

It would mean widgets that use the mixin would inherit from the latest changes in widgetBase automatically and reduce the overhead of keeping the two aligned.